test/src/e2e.dynval/dv.double-vote.test.ts

```diff
@@ -0,0 +1,476 @@
+    const encodableStep = (stepString => {
+        switch (stepString) {
+            case "propose":
+                return 0;
+            case "prevote":
+                return 1;
+            case "precommit":
+                return 2;
+            case "commit":
+                return 3;
+        }
+    })(step);
+    const encodableVoteStep = [
+        U64.ensure(height).toEncodeObject(),
+        U64.ensure(view).toEncodeObject(),
+        encodableStep
+    ];
+    const encodableBlockHash =
+        blockHash === null ? [] : [blockHash.toEncodeObject()];
+    const encodableVoteOn = [encodableVoteStep, encodableBlockHash];
+
+    const rlpVoteOn = RLP.encode(encodableVoteOn);
+    const messageForSchnorr = blake256(rlpVoteOn);
+    const schnorrSignature = signSchnorr(messageForSchnorr, privKey);
+    // pad because signSchnorr function does not guarantee the length of r and s to be 64.
+    const encodableSchnorrSignature = new H512(
+        schnorrSignature.r.padStart(64, "0") +
+            schnorrSignature.s.padStart(64, "0")
+    ).toEncodeObject();
+
+    return [
+        encodableVoteOn,
+        encodableSchnorrSignature,
+        U64.ensure(signerIdx).toEncodeObject()
+    ];
+}
+
+function createReportDoubleVoteTransaction(
+    sdk: SDK,
+    message1: RLP.Input,
+    message2: RLP.Input
+): Custom {
+    return sdk.core.createCustomTransaction({
+        handlerId: HANDLER_ID,
+        bytes: RLP.encode([REPORT_DOUBLE_VOTE_ACTION_ID, message1, message2])
+    });
+}
+
+const allDynValidators = originalValidators.slice(0, 4);
+const [alice, ...otherDynValidators] = allDynValidators;
+
+async function expectPossibleAuthors(
+    sdk: SDK,
+    expected: Signer[],
+    blockNumber?: number
+) {
+    const authors = (await stake.getPossibleAuthors(sdk, blockNumber))!.map(
+        author => author.toString()
+    );
+    expect(authors)
+        .to.have.lengthOf(expected.length)
+        .and.to.include.members(
+            expected.map(signer => signer.platformAddress.toString())
+        );
+}
+
+async function beforeBanCheck(sdk: SDK, blockNumber?: number) {
+    await expectPossibleAuthors(sdk, allDynValidators, blockNumber);
+}
+
+async function afterBanCheck(sdk: SDK, blockNumber?: number) {
+    await expectPossibleAuthors(sdk, otherDynValidators, blockNumber);
+    const bannedAfter = (await stake.getBanned(sdk, blockNumber)).map(
+        platformAddr => platformAddr.toString()
+    );
+    expect(bannedAfter).to.includes(alice.platformAddress.toString());
+    const delegteesAfter = (await stake.getDelegations(
+        sdk,
+        faucetAddress,
+        blockNumber
+    )).map(delegation => delegation.delegatee.toString());
+    expect(delegteesAfter).not.to.includes(alice.platformAddress.toString());
+}
+
+describe("Report Double Vote", function() {
+    const promiseExpect = new PromiseExpect();
+
+    async function getPossibleAuthorsByString(
```

Comment:
This is used only to get the index of Alice. Change this to `getAuthorIndex`

Reply:
I changed it.